lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -33,6 +33,22 @@ using namespace std;
 typedef DenseMap<Block *, vector<Value>> BlockValues;
 typedef DenseMap<Block *, vector<Operation *>> BlockOps;
 
+void removeRedundantComponents(handshake::FuncOp funcOp) {
+  for (auto &block : funcOp) {
+    for (auto &op : llvm::make_early_inc_range(block.getOperations())) {
+      if (auto mergeOp = dyn_cast<handshake::MergeOp>(op)) {
+        if (mergeOp.getNumOperands() == 1) {
+          mergeOp.getResult().replaceAllUsesWith(mergeOp.getOperand(0));
+          mergeOp.erase();
+        }
+      }
+      if (auto branchOp = dyn_cast<handshake::BranchOp>(op)) {
+        branchOp.getResult().replaceAllUsesWith(branchOp.getOperand());
+        branchOp.erase();
+      }
+    }
+  }
```

Comment:
Seems like this could be a canonicalizer associated with the ops instead?

Reply:
I found there are already some canonicalization patterns in the handshake dialect for removing redundant merge and branch operations. This patch only adds a test case for the canonicalization now.